bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/ContentVariableTransformer.java

```diff
@@ -0,0 +1,145 @@
+        SlingHttpServletRequest request = processingContext.getRequest();
+
+        contentVariableReplacements = aggregatorService.getProperties(request);
+    }
+
+    public void startElement(String uri, String localName, String quaName, Attributes atts) throws SAXException {
+        if (shouldRun()) {
+            AttributesImpl newAttrs = new AttributesImpl(atts);
+            for (int i = 0; i < newAttrs.getLength(); i++) {
+                String currentAttribute = newAttrs.getValue(i);
+                final List<String> keys = ContentVariableReplacementUtil.getKeys(currentAttribute);
+                for (String key : keys) {
+                    // If the placeholder key is in the map then replace it
+                    if (ContentVariableReplacementUtil.hasKey(contentVariableReplacements, key)) {
+                        String replaceValue =
+                                (String) ContentVariableReplacementUtil.getValue(contentVariableReplacements, key);
+                        String newAttrValue = ContentVariableReplacementUtil.doReplacement(currentAttribute, key,
+                                baseEscaping(replaceValue), propertyConfigService.getAction(key));
+                        newAttrs.setValue(i, newAttrValue);
+                    }
+                }
+            }
+            getContentHandler().startElement(uri, localName, quaName, newAttrs);
+        } else {
+            getContentHandler().startElement(uri, localName, quaName, atts);
+        }
+    }
+
+    public void characters(char[] ch, int start, int length) throws SAXException {
+        String currentString = new String(ch, start, length);
+
+        if (shouldRun()) {
+
+            // Get the current placeholders in the string
+            final List<String> keys = ContentVariableReplacementUtil.getKeys(currentString);
+            for (String key : keys) {
+                // If the placeholder key is in the map then replace it
+                if (ContentVariableReplacementUtil.hasKey(contentVariableReplacements, key)) {
+                    final String placeholderReplacement =
+                            String.valueOf(ContentVariableReplacementUtil.getValue(contentVariableReplacements, key));
+                    currentString = ContentVariableReplacementUtil.doReplacement(currentString, key,
+                            baseEscaping(placeholderReplacement), propertyConfigService.getAction(key));
+                }
+            }
+        }
+
+        getContentHandler().characters(currentString.toCharArray(), 0, currentString.length());
+    }
+
+    /**
+     * Applies the base level escaping unless otherwise overridden.
+     *
+     * @param input String to escape
+     * @return Escaped string
+     */
+    private String baseEscaping(String input) {
+        if (propertyConfigService.disableBaseEscaping()) {
```

Comment:
Do we truly want this option for people to turn off the escaping?  If they do this, they are opening up the ability for the feature to break their site at the whim of an author entering something that, when not escaped, will cause havoc.

Reply:
What would the case be?  Since it's a universal toggle, it feels very, very dangerous to me.

Reply:
Sure, but would it make better sense then to have the developer implement a `|html` action that avoids the escaping for that one value, rather than a universal toggle that removes the safety mechanism for all other cases?

Even if a developer is required to turn this on, it's authors that have the power to cause havoc, and the developer is left with no way to control it.  At least with a dedicated action an author has to explicitly express their desire to put unsafe content into the HTML body, whereas in the current state they can do it very accidentally and potentially even put the site in a state where they cannot fix the content depending on how badly the HTML of the page is broken (since the editor relies on the HTML being rendered).

Reply:
this has been updated - the 'disableEscaping' has been moved to the Actions